trial_balance.py

```diff
@@ -404,7 +404,8 @@ def _record(account, party, vals, comp, add_initial_balance):
         if comparison_fiscalyear:
             # second_dict = {}.fromkeys(accounts, Decimal('0.00'))
             logger.info('Calc initial vals for comparison period')
-            with transaction.set_context(periods=comparison_periods):
+            with transaction.set_context(fiscalyear=comparison_fiscalyear.id,
+                    periods=comparison_periods):
                 comparison_values = Account.read_account_vals(accounts,
```

Comment:
why add fiscalyear in the context? read_account_vals() has not get fiscalyer from the context... only the period [1]

[1] https://github.com/trytonspain/trytond-account_jasper_reports/blob/master/common.py#L84

Reply:
Because the fiscalyear is also ussed for getting the values of current values, so for coherence the same keys should be ussed for comparision values

Reply:
> Because the fiscalyear is also ussed for getting the values of current values, so for coherence the same keys should be ussed for comparision values

I review read_account_vals() and compute a query.  Could you do a reference where getting the values?

Reply:
Here it is used with fiscalyear:

https://github.com/trytonspain/trytond-account_jasper_reports/blob/master/trial_balance.py#L388

And here witout it:

https://github.com/trytonspain/trytond-account_jasper_reports/blob/master/trial_balance.py#L407

As both calls are equivalent but with diferent data they must have the same context keys

Reply:
I think the mistake is the line 388  because read_account_vals()  not get fiscalyear from the context. What do you think?

Reply:
Fixed in https://github.com/trytonspain/trytond-account_jasper_reports/pull/3/commits/3ee84e4d92098c8d677881eaa4207be4d4ecf413